src/ckanext-natcap/ckanext/natcap/plugin.py

```diff
@@ -244,51 +244,87 @@ def natcap_update_mappreview(context, package):
 
 
 def _load_download_rules_for(pkg):
-    """Try to get dataset-specific rules (first by title, then name)"""
+    """Try to get dataset-specific rules (first by title, then name)
+
+    Looks in dataset_configs folder for a file named after the dataset.
+
+    Args:
+        pkg (dict): CKAN package dictionary of metadata
+
+    Returns:
+        dict of parsed rules configuration or an empty dict if no
+        per-dataset config exists or loading fails.
+
+    """
     rules_dir = path.join(
         path.dirname(__file__), 'dataset_configs'
     )
     if not path.isdir(rules_dir):
         LOGGER.debug(f"dataset_configs not found at: {rules_dir}")
 
-    candidates = []
-    if pkg:
-        if pkg.get('title'):
-            candidates.append(path.join(rules_dir, f"{pkg['title']}.yml"))
-        if pkg.get('name'):
-            candidates.append(path.join(rules_dir, f"{pkg['name']}.yml"))
-            # name will be like: sts-12a3b4567 etc.
-
-    for fp in candidates:
-        try:
-            if path.exists(fp):
-                with open(fp, 'r') as f:
-                    data = yaml.safe_load(f) or {}
-                    LOGGER.info(f"Loaded per-dataset rules: {fp}")
-                    return data
-        except Exception:
-            LOGGER.exception(f"Failed loading per-dataset rules from {fp}")
+    potential_config_path = path.join(rules_dir, f"{pkg.get('title')}.yml")
+    try:
+        if path.exists(potential_config_path):
+            with open(potential_config_path, 'r') as f:
+                data = yaml.safe_load(f) or {}
```

Comment:
I don't think I commented on this before, but excellent idea to use `safe_load` here, even though we _should_ (hopefully!) be operating on trusted data.